data_catalog_backend/services/category_service.py

```diff
@@ -27,7 +27,16 @@ def get_category_by_title(self, title: str) -> Category:
 
     def get_categories(self) -> list[CategorySummaryResponse]:
         stmt = select(Category)
-        return self.session.scalars(stmt).all()
+        categories = self.session.execute(stmt).scalars().all()
+        return [
+            CategorySummaryResponse(
```

Comment:
You are mixing the responsibilites of the http-layer and the service-layer. 
CategorySummaryResponse is a type that should only be used at the router (http-layer). The service should return domain-objects (eg. the models) and the transformation to user objects (eg. the schemas) should happen at the router.

Ideally clean up the create_category as well (not changed in this PR, but has the same problem), so it doesn't rely on the schema-objects.

The one exception to this rule is the User-object.